UserService/UserService.Application/UseCases/AuthUseCases/EmailConfirmationUseCase/EmailConfirmationRequestHandler.cs

```diff
@@ -0,0 +1,32 @@
+﻿using MediatR;
+using Microsoft.AspNetCore.Identity;
+using System.Web;
+using UserService.Domain.Entities;
+using UserService.Domain.Exceptions;
+using UserService.Infrastructure.Services.EmailNotifications;
+
+namespace UserService.Application.UseCases.AuthUseCases.EmailConfirmationUseCase;
+
+internal class EmailConfirmationRequestHandler(UserManager<AppUser> userManager, IEmailService emailService)
+	: IRequestHandler<EmailConfirmationRequest>
+{
+	public async Task Handle(EmailConfirmationRequest request, CancellationToken cancellationToken)
+	{
+		var user = await userManager.FindByEmailAsync(request.email);
+		if (user is null)
+		{
+			throw new NotFoundException("No user with such email");
+		}
+
+		var code = HttpUtility.UrlDecode(request.code);
```

Comment:
Application layer have to know nothing about fact that you are using web. It is better to remove this from application layer